pkg/process/config/config.go

```diff
@@ -359,6 +363,10 @@ func NewAgentConfig(loggerName config.LoggerName, yamlPath, netYamlPath string)
 		cfg.LogLevel = "warn"
 	}
 
+	// get the hostname from the datadog agent
+	cfg.HostName = getHostnameFromGRPC(util.GetDDAgentClient)
```

Comment:
I wonder if it's better to return an error instead from `getHostnameFromGRPC` instead of just a string. `cfg.HostName, err = getHostnameFromGRPC(util.GetDDAgentClient)` and then fall back to the other hostname mechanism if `err != nil`

Reply:
Thanks for the suggestion. I've updated the function signature to return an error. I've intentionally left the conditional check in the fallback mechanism to check for the Hostname because the grpc call may return an empty string (i.e. when we are running on fargate).